### PR TITLE
 Create correct struct qty when size != dep.size 

### PR DESCRIPTION
**Pull request by sbechet**

### CODE REVIEW

File: YaLibs/YaToolsIDALib/IdaVisitor.cpp [hunk 1]
```diff
@@ -1199,9 +1199,24 @@ namespace
         make_views(version, ea);
     }
 
+    /* TODO: A day, create _one_ struct then array */
     bool try_apply_struc_to_data(ea_t ea, size_t size, const Tid& dep)
     {
-        auto ok = create_struct(ea, static_cast<asize_t>(size), dep.tid);
+        ea_t current_ea = ea;
+        int counter = 1;
+        auto ok = false;
```
Comment (bamiaux) [inline YaLibs/YaToolsIDALib/IdaVisitor.cpp:1207]: style: either use bool here, or auto on the two previous lines

File: YaLibs/YaToolsIDALib/IdaVisitor.cpp [hunk 2]
```diff
@@ -1199,9 +1199,24 @@ namespace
...
+
+        if ( (size != dep.size) && (size % dep.size == 0) )
```
Comment (bamiaux) [inline YaLibs/YaToolsIDALib/IdaVisitor.cpp:1209]: style: if(size != dep.size && size % dep.size)

File: YaLibs/YaToolsIDALib/IdaVisitor.cpp [hunk 3]
```diff
@@ -1199,9 +1199,24 @@ namespace
...
+        {
+            counter = size / dep.size;
+            do {
```
Comment (bamiaux) [inline YaLibs/YaToolsIDALib/IdaVisitor.cpp:1212]: style: avoid "do {} while" when possible
```
for(size_t i = 0; ok && i < size; i += dep.size)
    ok = create_struct(ea + i , static_cast_<asize_t>(dep.size), dep.tid);
```

**Review by bamiaux — COMMENTED** [review]
It's missing a test

**Comment by sbechet** [comment]
Ok, thank you.

I will make the corrections and add the tests I will come back with a new pull request.

**Pull request closed without merging**